app/tests/components_tests/test_tasks.py

```diff
@@ -422,3 +423,49 @@ def test_add_file_to_object(settings, object_type):
         interface_pk=ci.pk,
     )
     assert ComponentInterfaceValue.objects.filter(interface=ci).count() == 1
+
+
+class RandomException(Exception):
+    pass
+
+
+@pytest.mark.django_db
+@mock.patch(
+    "grandchallenge.uploads.models.UserUpload.copy_object",
+    side_effect=RandomException,
+)
+@pytest.mark.parametrize(
+    "object_type", [DisplaySetFactory, ArchiveItemFactory]
+)
+def test_add_file_to_object_exception_handling(settings, object_type):
+    settings.task_eager_propagates = (True,)
+    settings.task_always_eager = (True,)
+
+    creator = UserFactory()
+    obj = object_type()
+
+    us = UserUploadFactory(filename="file.json", creator=creator)
+    presigned_urls = us.generate_presigned_urls(part_numbers=[1])
+    response = put(presigned_urls["1"], data=b'{"foo": "bar"}')
+    us.complete_multipart_upload(
+        parts=[{"ETag": response.headers["ETag"], "PartNumber": 1}]
+    )
+    us.save()
+
+    ci = ComponentInterfaceFactory(
+        kind="JSON",
+        store_in_database=False,
+    )
+
+    with pytest.raises(RandomException):
+        add_file_to_object(
+            app_label=obj._meta.app_label,
+            model_name=obj._meta.model_name,
+            object_pk=obj.pk,
+            user_upload_pk=us.pk,
+            interface_pk=ci.pk,
+        )
+
+    notification = Notification.objects.last()
```

Comment:
This test fails and I cannot work out why. I stepped through the code and verified that the notification is created, but it is not visible in the test. I think this has to do with different DB transactions, but I cannot figure out what needs to change in the test for the notification to be visible.

Reply:
I can't immediately see it. Honestly, I don't think we really need to solve this issue, we have the error in Sentry and investigate each occurrence anyway. Probably best to move on and close this PR as it's not straightforward.

Reply:
Ok. I agree.  It shouldn't happen very often anyway. And if we encounter issues more frequently, we can revisit this.